proposals/infra/INF-0002-root-signature-in-clang.md

```diff
@@ -304,15 +304,147 @@ elimation has completed.
 
 ### Validations in Sema
 
-TODO
+#### All the values should be legal.
+
+  ShaderVisibility
+
+      'SHADER_VISIBILITY_ALL'
+      'SHADER_VISIBILITY_VERTEX'
+      'SHADER_VISIBILITY_HULL'
+      'SHADER_VISIBILITY_DOMAIN'
+      'SHADER_VISIBILITY_GEOMETRY'
+      'SHADER_VISIBILITY_PIXEL'
+      'SHADER_VISIBILITY_AMPLIFICATION'
+      'SHADER_VISIBILITY_MESH'
+
+  ParameterType
+
+      CBV
+      SRV
+      UAV
+      DescriptorTable
+      Constants32Bit
+
+  RootDescriptorFlags
+
+      None
+      DataVolatile
+      DataStaticWihleSetAtExecute
+      DataStatic
+
+  DescriptorRangeFlags
+
+    For Sampler
+      None
+      DescriptorsVolatile
+      DESCRIPTORS_STATIC_KEEPING_BUFFER_BOUNDS_CHECKS
+
+    For resource
```

Comment:
Is this saying that not all combinations of flags for resources are valid?  If so, then this should also be validated in codegen.

Reply:
In fact, I think that in any cases where the metadata allows invalid root signatures to be represented we need to validate them again during codegen.  Codegen cannot assume that the metadata it has been given is valid.

Reply:
Yes. Many of these value checks are covered by syntax check in Sema, 
list all of these for DXIL generation check to reference.